### ISSUE TITLE
[Deque Analysis] Form elements, tabs, buttons & dropdowns have insufficient color contrast

### ISSUE THREAD

**Issue by tdonohue** [DSpace/dspace-angular#1151]
## Deque Analysis Summary

This issue exists on 8 pages, spread across 43 issues in our Deque analysis.  In all situations, a form element has the following color contrast issue:

"Element has insufficient color contrast of 4.47 (foreground color: `#777777`, background color: `#ffffff`). Contrast must be 4.5 or more."

1. Search Results (9 tickets)
   * https://demo7.dspace.org/search?spc.page=1&query=test&spc.sf=score&spc.sd=DESC
   * **Locations:** 
       * All Text Input Fields ("Search the repository", "Author Name", "Subject", "Date" fields, "Item Type"). Outline/color is too light
       * All dropdown/selectbox fields ("All repository" "Sort By", "Results per page"). Outline/color is too light
   * Deque issue ticket IDs: 468500, 468501, 468502, 468507, 468508, 468509, 468510, 468511, 468928
2. Login module (2 tickets)
    * (on all pages)
    * **Locations:** Text input field for email address/password. Outline/color is too light
    * Deque issue ticket IDs: 469225, 469226
3. Browse by Author (1 ticket)
    * https://demo7.dspace.org/browse/author
    * **Locations:** Text input field for "Type the first few letters.." box. Outline/color is too light
    * Deque issue ticket IDs: 469336
4. Homepage (1 ticket)
    * https://demo7.dspace.org/home
    * **Locations:** Text input field "Search the repository.." box. Outline/color is too light
    * Deque issue ticket IDs: 469469
5. Submission (12 tickets)
    * https://demo7.dspace.org/submit
    * **Locations:** All input fields. Outline/color is too light
    * Deque issue ticket IDs: 469506, 469508, 469510, 469512, 469514, 469516, 469518, 469519, 469521, 469524, 469526, 469528
6. Edit Item (6 tickets)
    * https://demo7.dspace.org/entities/publication/2f4ec582-109e-4952-a94a-b7d7615a8c69/edit/status
    * **Locations:** 
        * "Status" tab: Color of selected tab is too light.  Color of "Back" button is too light.
        * "Metadata" tab: Color of selected tab is too light.  Color of "Back" button is too light.  Click edit on a row...the displayed text input fields have a color/outline that is too light.
        * "Bitstreams" tab: Color of selected tab is too light.  Color of "Back" button is too light.
    * Deque issue ticket IDs: 470072, 470073, 470727, 470729, 470738, 471445
 7. Edit Collection (9 tickets)
     * https://demo7.dspace.org/collections/0f58a9e3-254a-4a38-a6ea-8c1922762475/edit/metadata
     * **Locations:** Color of selected tab is too light.  Color of "Back" button is too light.  Color/Outline of all input fields is too light.
     * Deque issue ticket IDs: 470080, 470089, 470082, 470083, 470084, 470085, 470086, 470087, 470088
 8. MyDSpace (3 tickets)
     * https://demo7.dspace.org/mydspace?configuration=workspace&spc.sf=score&spc.sd=DESC
     * **Locations:** Color of all dropdowns (Show, All Repository, Sort By, Results per page) is too light.  Color/Outline of all input fields (in filters & "Search in mydspace") is too light.
     * Deque issue ticket IDs: 470116, 470121, 470123

## Recommended Fix

All of these are fixed by simply making the default color of our form fields *darker*.  It looks like many of these *might* be controlled by our `form-control` class, which is inheriting the `#777777` from somewhere.

We should find a way to make all form fields/tabs/buttons darker in color, minimally `#757575` (but darker than that is also OK).

## More Information / Tools

* As necessary, see mentioned Issue IDs in our assessment results (requires login): https://axeauditor.dequecloud.com/test-run/0856438a-a19a-11eb-bc31-b7d5be387c86/issues
* Consider using [Deque's free Chrome plugin](https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US) to check your work
* Deque's color contrast tool: https://dequeuniversity.com/rules/axe/4.1/color-contrast?application=axeAPI

**Comment by tdonohue** [comment]
@atarix83 : Assigning this to you for you/your team to analyze, as I know you've done a lot of work with form fields (especially in submission).  Please keep in mind though that this ticket only involves fixing the color of form fields...you should not worry about fixing the color of their "placeholder" text, as that will be handled separately in #1154.

**Comment by atarix83** [comment]
I think can be resolved ina 1 hour and I suggest to do it within the #1154 in order to resolve them by using the same approach

**Comment by tdonohue** [comment]
@artlowel : As you can see above, @atarix83 has noted this ticket might be most easily solved alongside #1154.  As you are assigned to that ticket, does that sound reasonable to you as well?  Both of these are relatively small effort (though they affect a large number of fields).

For now, thanks for the estimate @atarix83 .  I'll add the estimate but **unassign** this ticket until we clarify who will work on it.

**Comment by artlowel** [comment]
Yes, most of these will likely be fixed by the solution to #1154
